okapi-core/src/main/java/org/folio/okapi/managers/ProxyService.java

```diff
@@ -954,19 +959,38 @@ private void proxyInternal(Iterator<ModuleInstance> it,
     );
   }
 
+  static Buffer compressInternalResponse(RoutingContext ctx, Buffer respBuf) {
+    String acceptEncoding = ctx.request().getHeader("Accept-Encoding");
+    int sz = respBuf.length();
+    if (acceptEncoding != null && acceptEncoding.contains("gzip")
+        && sz > CONTENT_COMPRESS_MIN_SIZE) {
+      try {
+        ByteArrayOutputStream dst = new ByteArrayOutputStream();
```

Comment:
Seems like `ByteArrayOutputStream` has a `close` method. I haven't looked at its source but just googling on it there seems to be a difference of opinion on whether to call it does anything since the implementation may not do anything. FWIW someone seems to think it's still a good idea to call in case the implementation changes in future releases. Also FWIW unclosed streams are bad in my experience. Could be the same for the stream that's doing the zipping.